docs/source/user-guide/sql/ddl.md

```diff
@@ -47,17 +47,50 @@ CREATE SCHEMA cat.emu;
 
 ## CREATE EXTERNAL TABLE
 
-Parquet data sources can be registered by executing a `CREATE EXTERNAL TABLE` SQL statement. It is not necessary
-to provide schema information for Parquet files.
+`CREATE EXTERNAL TABLE` SQL statement registers a location on a local
+file system or remote object store as a named table which can be queried.
+
+The supported syntax is:
+
+```
+CREATE EXTERNAL TABLE
+[ IF NOT EXISTS ]
+<TABLE_NAME>[ (<column_definition>) ]
+STORED AS <file_type>
+[ WITH HEADER ROW ]
+[ DELIMITER <char> ]
+[ COMPRESSION TYPE <GZIP | BZIP2 | XZ | ZSTD> ]
+[ PARTITIONED BY (<column list>) ]
+[ WITH ORDER (<ordered column list>)
+[ OPTIONS (<key_value_list>)
+LOCATION <literal>
+
+<column_definition> := (<column_name> <data_type>, ...)
+
+<column_list> := (<column_name>, ...)
+
+<ordered_column_list> := (<column_name> <sort_clause>, ...)
+
+<key_value_list> := (<literal> <literal, <literal> <literal>, ...)
```

Comment:
Is there a collection of available options? If someone were to implement a mechanism based on options, how should it be documented?

Reply:
> If someone were to implement a mechanism based on options, how should it be documented?

I recommend anything meant to be used by users of datafusion should be explicitly in the `CREATE TABLE` syntax -- e.g. https://github.com/apache/arrow-datafusion/issues/6248)

If we do want to do something with options, perhaps it could be documented in https://arrow.apache.org/datafusion/user-guide/sql/ddl.html